engine/src/main/java/org/terasology/rendering/dag/stateChanges/DisableMaterial.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.stateChanges;
+
+import org.terasology.registry.CoreRegistry;
+import org.terasology.rendering.ShaderManager;
+import org.terasology.rendering.dag.StateChange;
+
+/**
+ * TODO: Add javadocs
+ */
+public final class DisableMaterial implements StateChange {
+    private ShaderManager shaderManager = CoreRegistry.get(ShaderManager.class);
```

Comment:
Hmm, this is a tricky one. The EnableMaterial() originally got it from the CoreRegistry too, and since the DisableMaterial is a static object in class EnableMaterial, it's kinda hard..

Right now, I don't think I can do it easily. Just adding a comment for now, but I'll try to make the entire DAG stop relying on CoreRegistry in the near future.